Mathlib/Analysis/Meromorphic/TrailingCoefficient.lean

```diff
@@ -0,0 +1,258 @@
+  by_cases h₁ : MeromorphicAt f x
+  · by_cases h₂ : meromorphicOrderAt f x = ⊤
+    · exact 0
+    · exact ((meromorphicOrderAt_ne_top_iff h₁).1 h₂).choose x
+  · exact 0
+
+/--
+If `f` is not meromorphic at `x`, the trailing coefficient is zero by definition.
+-/
+@[simp] lemma meromorphicTrailingCoeffAt_of_not_MeromorphicAt (h : ¬MeromorphicAt f x) :
+    meromorphicTrailingCoeffAt f x = 0 := by simp_all [meromorphicTrailingCoeffAt]
+
+/--
+If `f` is meromorphic of infinite order at `x`, the trailing coefficient is zero by definition.
+-/
+@[simp] lemma MeromorphicAt.meromorphicTrailingCoeffAt_of_order_eq_top (h₁ : MeromorphicAt f x)
+    (h₂ : meromorphicOrderAt f x = ⊤) :
+    meromorphicTrailingCoeffAt f x = 0 := by simp_all [meromorphicTrailingCoeffAt]
+
+/-!
+## Characterization of the Leading Coefficient
+-/
+
+/--
+Definition of the trailing coefficient in case where `f` is meromorphic of finite order and a
+presentation is given.
+-/
+@[simp]
+lemma MeromorphicAt.meromorphicTrailingCoeffAt_of_order_eq_finite (h₁ : MeromorphicAt f x)
+    (h₂ : AnalyticAt 𝕜 g x) (h₃ : meromorphicOrderAt f x ≠ ⊤)
+    (h₄ : f =ᶠ[𝓝[≠] x] fun z ↦ (z - x) ^ (meromorphicOrderAt f x).untop₀ • g z) :
+    meromorphicTrailingCoeffAt f x = g x := by
+  unfold meromorphicTrailingCoeffAt
+  simp only [h₁, not_true_eq_false, reduceDIte, h₃, ne_eq]
+  obtain ⟨h'₁, h'₂, h'₃⟩ := ((meromorphicOrderAt_ne_top_iff h₁).1 h₃).choose_spec
+  apply Filter.EventuallyEq.eq_of_nhds
+  rw [← h'₁.continuousAt.eventuallyEq_nhds_iff_eventuallyEq_nhdsNE h₂.continuousAt]
+  filter_upwards [h₄, h'₃, self_mem_nhdsWithin] with y h₁y h₂y h₃y
+  rw [← sub_eq_zero]
+  rwa [h₂y, ← sub_eq_zero, ← smul_sub, smul_eq_zero_iff_right] at h₁y
+  simp_all [zpow_ne_zero, sub_ne_zero]
+
+/--
+Variant of `meromorphicTrailingCoeffAt_of_order_eq_finite`: Definition of the trailing coefficient
+in case where `f` is meromorphic of finite order and a presentation is given.
+-/
+@[simp]
+lemma AnalyticAt.meromorphicTrailingCoeffAt_of_order_eq_finite₁ (h₁ : AnalyticAt 𝕜 g x)
+    (h₂ : g x ≠ 0) (h₃ : f =ᶠ[𝓝[≠] x] fun z ↦ (z - x) ^ n • g z) :
+    meromorphicTrailingCoeffAt f x = g x := by
+  have h₄ : MeromorphicAt f x := by
+    rw [MeromorphicAt.meromorphicAt_congr h₃]
+    fun_prop
+  have : meromorphicOrderAt f x = n := by
+    simp only [meromorphicOrderAt_eq_int_iff h₄, ne_eq, zpow_natCast]
+    use g, h₁, h₂
+    exact h₃
+  simp_all [h₄.meromorphicTrailingCoeffAt_of_order_eq_finite h₁, this]
+
+/--
+If `f` is analytic and does not vanish at `x`, then the trailing coefficient of `f` at `x` is `f x`.
+-/
+@[simp]
+lemma AnalyticAt.meromorphicTrailingCoeffAt_of_nonvanish (h₁ : AnalyticAt 𝕜 f x) (h₂ : f x ≠ 0) :
+    meromorphicTrailingCoeffAt f x = f x := by
+  rw [h₁.meromorphicTrailingCoeffAt_of_order_eq_finite₁ (n := 0) h₂]
+  filter_upwards
+  simp
+
+/--
+If `f` is meromorphic at `x`, then the trailing coefficient of `f` at `x` is the limit of the
+function `(· - x) ^ (-h₁.order.untop₀) • f`.
```

Comment:
Can you fix the docstring, not mentioning `h₁.order`?

Reply:
Implemented. Just wrote "order", similar to what I did elsewhere.